test/framework/test_composite.py

```diff
@@ -151,6 +152,29 @@ def test_roundtrip_serializable(self):
 
         self.assertRoundTripSerializable(exp)
 
+    @data(None, "Test")
+    def test_experiment_type(self, exp_type):
+        """Test the experiment_type setter for the parallel experiment and
```

Comment:
Once you remove the setters from these classes, you can also remove the test here. However do test the the init arg (such a test is currently missing).